### DISCUSSION TITLE
Livewire 3 equivalent for Livewire 2 'livewire:update' hook

### DISCUSSION THREAD

**Discussion by KeanuTang** [livewire/livewire#8265] [1. Help]
Hi all,

I'm upgrading my application from Livewire 2 to Livewire 3. If memory serves me right, the 'livewire:update' hook in Livewire 2 would fire after a network response and DOM manipulation is done. In Livewire 3, I see a couple of alternatives, but they aren't ideal.

- 'morph.updated' - This fires after the DOM is updated, but it fires after every DOM manipulation. Ideally I'd like to fire my hook only once per network request. 
- 'request' > succeed - This fires once per network request, but it seems to fire before the DOM is updated.

Is there another, more appropriate hook? I'm resorting to 'morph.updated' because I have to fire my code after the DOM is updated, but I'm worried about the performance impact of firing it after every individual DOM update. 

Thanks!

**Comment by PerryvanderMeer** [comment]
How about the [commit hooks](https://livewire.laravel.com/docs/javascript#commit-hooks)?
You can hook into `succeed`, if necessary combined with Alpine's [$nextTick](https://alpinejs.dev/magics/nextTick).

**Reply by KeanuTang** [reply]
Thanks Perry! That's a good idea. I wasn't aware of nextTick. I can use that with either the commit or response hooks it seems.

It still would be nice to have something a bit more built-in, but this does seem to accomplish what I'm trying to do:

```
$(document).ready(function () {
    Livewire.hook('request', ({ component, commit, respond, succeed, fail }) => {
        succeed(({ snapshot, effect }) => {
            Alpine.nextTick(() => { processDOMChanges(); });
        })
    })
});
```

**Reply by joshhanley** [reply]
@KeanuTang `succeed()` should work without `$nextTick()` as it is fired after the morph and everything is complete. But I do believe it happens before the browser repaints, hence you would need `$nextTick()` if you are working on the visible DOM.

**Reply by KeanuTang** [reply]
Thanks @joshhanley . I think that's my issue. I'm trying to process the returned HTML to add certain JavaScript click events. It doesn't seem to be working until the DOM changes are actually rendered for some reason.

**Comment by fjeansson** [comment]
We really need this built in...

**Reply by KeanuTang** [reply]
I agree. I accomplished it with the suggestions in this post, but it should be more native than what I had to do.

**Reply by lrljoe** [reply]
What exactly are you trying to do, as there will probably be a way to do it without needing to hook into dom events 

@KeanuTang you mentioned you were trying to add some click actions, can you be more specific?  It should be relatively simple to do that kind of thing natively without modifying the DOM after it is rendered.

It's difficult to advise based on so little information about your use case.

**Reply by KeanuTang** [reply]
It's been a while since I've worked on this project. We had some jquery scripts that would normally fire to wire up buttons to do particular actions. The problem we were running into is, after a livewire update, new buttons could be added to the page, but the jquery scripts were not trigger and so the buttons were not wired up.

Basically we had an onload event triggering: $(selector).on("click", functionToFindAndWireUpButtons)

I can dig up some more details if it's helpful, but that seems like that should give you an idea of what we were doing.
